app/actions/remote/entry.ts

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Model} from '@nozbe/watermelondb';
+
+import {scheduleExpiredNotification} from '@actions/local/push_notification';
+import {General, Preferences} from '@constants';
+import DatabaseManager from '@database/manager';
+import {getPreferenceValue, getTeammateNameDisplaySetting} from '@helpers/api/preference';
+import {selectDefaultTeam} from '@helpers/api/team';
+import NetworkManager from '@init/network_manager';
+import {prepareMyChannelsForTeam} from '@queries/servers/channel';
+import {prepareMyPreferences} from '@queries/servers/preference';
+import {prepareCommonSystemValues} from '@queries/servers/system';
+import {addChannelToTeamHistory, prepareMyTeams} from '@queries/servers/team';
+import {selectDefaultChannelForTeam} from '@utils/channel';
+
+import {fetchMissingSidebarInfo, fetchMyChannelsForTeam, MyChannelsRequest} from './channel';
+import {fetchPostsForChannel, fetchPostsForUnreadChannels} from './post';
+import {MyPreferencesRequest, fetchMyPreferences} from './preference';
+import {fetchRolesIfNeeded} from './role';
+import {ConfigAndLicenseRequest, fetchConfigAndLicense} from './systems';
+import {fetchMyTeams, fetchTeamsChannelsAndUnreadPosts, MyTeamsRequest} from './team';
+
+import type {Client} from '@client/rest';
+
+type AfterLoginArgs = {
+    serverUrl: string;
+    user: UserProfile;
+    deviceToken?: string;
+}
+
+export const loginEntry = async ({serverUrl, user, deviceToken}: AfterLoginArgs) => {
+    const dt = Date.now();
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    if (deviceToken) {
+        try {
+            client.attachDevice(deviceToken);
+        } catch {
+            // do nothing, the token could've failed to attach to the session but is not a blocker
```

Comment:
Will it try to re-attach at a later time or on another flow ?

Reply:
While logging in, this would only cause that SSO login sessions could not attach the device Id, attaching the device Id should also occur for the normal entry point logic (aka: opening the app from the home screen)